src/main/java/de/srendi/advancedperipherals/common/addons/minecolonies/MineColonies.java

```diff
@@ -0,0 +1,367 @@
+     * Converts a home {@link IBuilding} to a map
+     *
+     * @param home the home building
+     * @return a map with information about the home building
+     */
+    public static Object homeToObject(IBuilding home) {
+        Map<String, Object> map = new HashMap<>();
+        map.put("location", LuaConverter.posToObject(home.getLocation().getInDimensionLocation()));
+        map.put("type", home.getSchematicName());
+        map.put("level", home.getBuildingLevel());
+
+        return map;
+    }
+
+    /**
+     * Converts a skill {@link Skill} into a map
+     *
+     * @param skills skills as list. Can be obtained via {@link ICitizenData#getCitizenSkillHandler}
+     * @return a map with information about the skill
+     */
+    public static Object skillsToObject(Map<Skill, CitizenSkillHandler.SkillData> skills) {
+        Map<String, Object> map = new HashMap<>();
+        for (Skill skill : skills.keySet()) {
+            Map<String, Object> skillData = new HashMap<>();
+            skillData.put("level", skills.get(skill).getLevel());
+            skillData.put("xp", skills.get(skill).getExperience());
+            map.put(skill.name(), skillData);
+        }
+
+        return map;
+    }
+
+    /**
+     * Returns information about the building like structure data, the citizens and some other values
+     *
+     * @param buildingManager The building manager of the colony
+     * @param building        The building as instance
+     * @param pos             The location of the buildings block
+     * @return information about the building
+     */
+    public static Object buildingToObject(IRegisteredStructureManager buildingManager, IBuilding building, BlockPos pos) {
+        Map<String, Object> structureData = new HashMap<>();
+        structureData.put("cornerA", LuaConverter.posToObject(building.getCorners().getA()));
+        structureData.put("cornerB", LuaConverter.posToObject(building.getCorners().getB()));
+        structureData.put("rotation", building.getRotationMirror().toString());
+        structureData.put("mirror", building.getRotationMirror().isMirrored());
+
+        List<Object> citizensData = new ArrayList<>();
+        for (ICitizenData citizen : building.getAllAssignedCitizen()) {
+            Map<String, Object> citizenData = new HashMap<>();
+            citizenData.put("id", citizen.getId());
+            citizenData.put("name", citizen.getName());
+            citizensData.add(citizenData);
+        }
+
+        Map<String, Object> map = new HashMap<>();
+        map.put("location", LuaConverter.posToObject(pos));
+        map.put("type", building.getSchematicName());
+        map.put("style", building.getStructurePack());
+        map.put("level", building.getBuildingLevel());
+        map.put("maxLevel", building.getMaxBuildingLevel());
+        map.put("name", building.getBuildingDisplayName());
+        map.put("built", building.isBuilt());
+        map.put("isWorkingOn", building.hasWorkOrder());
```

Comment:
`"isWorking"` makes more sense here I think. `isWorkingOn` means you have the id you are working?